### PR TITLE
Migrate forging component to React - Closes #351

### LINKED ISSUE #351: Migrate Forging tab to React

### PR DESCRIPTION
**Pull request by slaweet**
Closes #351

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,10 +35,12 @@
     "postcss-cssnext": "=2.11.0",
     "prop-types": "=15.5.10",
     "react": "=15.6.x",
+    "react-circular-progressbar": "=0.1.5",
     "react-dom": "=15.6.x",
     "react-redux": "=5.0.3",
     "react-redux-toastr": "=7.0.0",
     "react-router-dom": "=4.0.0",
+    "react-timeago": "=3.3.0",
```
Comment (yasharAyari) [inline package.json]: Please remove this dependency and use our custom timestamp component

File: src/store/reducers/forging.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  let endTimesamp;
+
+  switch (action.type) {
+    case actionTypes.forgedBlocksUpdated:
+      startTimesamp = state.forgedBlocks.length ? state.forgedBlocks[0].timestamp : 0;
+      endTimesamp = state.forgedBlocks.length ?
+        state.forgedBlocks[state.forgedBlocks.length - 1].timestamp :
+        0;
+      return Object.assign({}, state, {
+        forgedBlocks: [].concat(
```
Comment (yasharAyari) [inline src/store/reducers/forging.js]: Replace [].concat with es6 spread syntax. You can find more about spread syntax here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

File: src/components/forging/delegateStats.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import { mount } from 'enzyme';
+import DelegateStats from './delegateStats';
+
+chai.use(sinonChai);
+
+describe('<DelegateStats />', () => {
```
Comment (reyraa) [inline src/components/forging/delegateStats.test.js]: Please remove html annotation

File: src/components/forging/forgedBlocks.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import { mount } from 'enzyme';
+import ForgedBlocks from './forgedBlocks';
+
+chai.use(sinonChai);
+
+describe('<ForgedBlocks />', () => {
```
Comment (reyraa) [inline src/components/forging/forgedBlocks.test.js]: Please remove html annotation

File: src/components/forging/forgingStats.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import { mount } from 'enzyme';
+import ForgingStats from './forgingStats';
+
+chai.use(sinonChai);
+
+
+describe('<ForgingStats />', () => {
```
Comment (reyraa) [inline src/components/forging/forgingStats.test.js]: Same here

File: src/components/forging/forgingTitle.test.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import { mount } from 'enzyme';
+import ForgingTitle from './forgingTitle';
+
+chai.use(sinonChai);
+
+
+describe('<ForgingTitle />', () => {
```
Comment (reyraa) [inline src/components/forging/forgingTitle.test.js]: same here

File: src/components/forging/index.js [hunk 7]
```diff
@@ -1,7 +1,31 @@
...
-);
+const mapStateToProps = state => ({
+  account: state.account,
+  peers: state.peers,
+  statistics: state.forging.statistics,
+  forgedBlocks: state.forging.forgedBlocks,
+});
+
+const mapDispatchToProps = dispatch => ({
+  loadForgedBlocks: (activePeer, limit, offset, generatorPublicKey) => {
```
Comment (reyraa) [inline src/components/forging/index.js]: To keep our code style consistent, Please follow the approach in login, transaction and account components by moving the Api call to the component consuming the data, and only dispatching the data here.

File: src/components/forging/index.js [hunk 8]
```diff
@@ -1,7 +1,31 @@
...
+    getForgedBlocks(activePeer, limit, offset, generatorPublicKey).then((data) => {
+      dispatch(updateForgedBlocks(data.blocks));
+    });
+  },
+  loadStats: (activePeer, key, startMoment, generatorPublicKey) => {
+    getForgedStats(activePeer, startMoment, generatorPublicKey).then((data) => {
```
Comment (reyraa) [inline src/components/forging/index.js]: same here

**Pull request merged**